http-clients/apache-client/src/main/java/software/amazon/awssdk/http/apache/ApacheHttpClient.java

```diff
@@ -304,6 +311,27 @@ default Builder apply(Consumer<Builder> mutator) {
          * Configure the maximum amount of time that a connection should be allowed to remain open while idle.
          */
         Builder connectionMaxIdleTime(Duration maxIdleConnectionTimeout);
+
+        /**
+         * Forces the client to trust all certificates, even invalid or self signed certificates. This should only ever
+         * be used for testing purposes.
+         *
+         * @param trustAllCertificates Whether to trust all certificates. The default is false and only valid certificates
+         *                             whose trust can be verified via the trust store will be trusted.
+         * @return This builder for method chaining.
+         */
+        Builder trustAllCertificates(Boolean trustAllCertificates);
```

Comment:
So you don't want to include an option for HTTP to promote security but we are okay with this ;)

Reply:
There is a workaround to use Http endpoint directly via client. But there is no option to disable this for testing. Ideally I don't want to expose it too if we have a easy way to test by overriding certs.

Reply:
We can leave it as a system property like we do for V1 and resolve that in the defaults we pass in. That way we don't need to expose an option in every http impl we support, they would just be able to read it from the attribute map.